src/org/labkey/test/components/bootstrap/LoggedOutDialog.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public void clickReloadPage()
+    {
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().reloadPageBtn));
+        elementCache().reloadPageBtn.click();
+    }
+
+    public void reloadPageAndConfirmPostLoginRedirect(String expectedURL)
+    {
+        clickReloadPage();  // dismisses the dialog
+
+        // at this point, we're either at a login page already, or at a public page showing a link to one
+        // when we add biologics to this, there will likely be a third case to handle here
+        WebElement el = Locator.waitForAnyElement(new FluentWait<SearchContext>(getDriver()).withTimeout(Duration.ofMillis(WAIT_FOR_JAVASCRIPT)),
+                org.labkey.test.Locators.signInLink,
+                Locator.tagWithName("form", "login"));
+
+        if (el.getAttribute("class").equals("header-link"))
+            getWrapper().clickAndWait(el);
+
+        // We should now be on the regular login page, without the beginAt action simpleSignIn does
+        getWrapper().assertElementPresent(Locator.tagWithName("form", "login"));
+        getWrapper().setFormElement(Locator.name("email"), PasswordUtil.getUsername());
+        getWrapper().setFormElement(Locator.name("password"), PasswordUtil.getPassword());
+        getWrapper().clickButton("Sign In", 0);
+
+        var returnUrl = getWrapper().getCurrentRelativeURL();
+        assertThat("Signing back in did not take us back to the expected page.", returnUrl, is(expectedURL));
+    }
+
+    /**
+     *  Checks the background/masked elements behind the dialog to ensure that the expected content
+     *  areas are blurred/styled as expected.
+     *  Because not all of these page sections are present in all situations (notably, in apps like
+     *  SM, FM, Biologics) this checks for the ones present, and if any of them are both present and
+     *  not styled as expected, will return false.
+     * @return True if all page sections that are present are styled with 'lk-content-blur', otherwise false
+     */
+    public boolean isContentBlurred()
+    {
+        List<Boolean> factors = new ArrayList<>();
+        var lkHeader = elementCache().lkHeader.findOptionalElement(getDriver());
+        if (lkHeader.isPresent())
+            factors.add(lkHeader.get().getAttribute("class").contains("lk-content-blur"));
+
+        var lkBody = elementCache().lkBody.findOptionalElement(getDriver());
+        if (lkBody.isPresent())
+            factors.add(lkBody.get().getAttribute("class").contains("lk-content-blur"));
+
+        var footerBlock = elementCache().footerBlock.findOptionalElement(getDriver());
+        if (footerBlock.isPresent())
+            factors.add(footerBlock.get().getAttribute("class").contains("lk-content-blur"));
+
```

Comment:
These checks are not reliable; if the header/body/footer Locators become invalid we would never know whether they were being properly blurred. I'd say to have this take a list of Locators for things that should be blurred and have this method be strict about the elements being present. Also, why not return immediately as soon as we find a non-blurred element?
```suggestion
    public boolean isContentBlurred(Locator... shouldBeBlurred)
    {
        for (Locator loc : shouldBeBlurred)
        {
            if (!loc.findElement(getDriver()).getAttribute("class").contains("lk-content-blur"))
                return false;
        }
        return true;
```

Reply:
Header and footer aren't present in react Apps (like SM, FM) so this logic wouldn't work for this feature in them.
I could write separate methods for checking in-app vs. in LKS that could be strict in this way
Or I could write a method that requires them if we're in an app
...I consolidated the code to work in these cases after my first approach (verify different situations separately) was rejected in favor of not duplicating code, so at this point you people are hard to please hahaha

Reply:
I've implemented the changes we discussed offline; now the helper takes a list of Locator of elements to check for styling and will fail if they aren't present.  I've updated the tests to specify the top-level elements expected to be present (for the app, or for LKS) and added an additional check to make sure we know it if these top-level element classes are changed out from under us